v2/blockstore/readwrite.go

```diff
@@ -354,6 +378,11 @@ func (b *ReadWrite) Discard() {
 // for more efficient subsequent read.
 // After this call, the blockstore can no longer be used.
 func (b *ReadWrite) Finalize() error {
+	if b.opts.WriteAsCarV1 {
+		b.Discard()
```

Comment:
I'd document here why we have nothing to do in the v1 case. For example, right now I forget why Finalize doesn't need to write the carv1 data at all - is it already written to disk?

Reply:
Yeah, already written to disk, no trailer needed for a v1. I'm also changing to a direct `b.ronly.Close()` call instead of `b.Discard()` to be less confusing.